src/ui/blockUi-editor.js

```diff
@@ -114,6 +118,145 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    // a block editor that includes a progress indicator
+    fluid.defaults("sjrk.storyTelling.blockUi.editor.withFileUploader", {
+        gradeNames: ["sjrk.storyTelling.blockUi.editor"],
+        model: {
+            previewVisible: true,
+            progressAreaVisible: false,
+            responseAreaVisible: false,
+            uploadButtonDisabled: false
+        },
+        modelListeners: {
+            previewVisible: {
+                this: "{that}.dom.mediaPreview",
+                method: "toggle",
+                args: ["{change}.value"],
+                namespace: "previewVisibleChange"
+            },
+            progressAreaVisible: {
+                this: "{that}.dom.progressArea",
+                method: "toggle",
+                args: ["{change}.value"],
+                namespace: "progressAreaVisibleChange"
+            },
+            responseAreaVisible: {
+                this: "{that}.dom.responseArea",
+                method: "toggle",
+                args: ["{change}.value"],
+                namespace: "responseAreaVisibleChange"
+            },
+            uploadButtonDisabled: {
+                this: "{that}.dom.uploadButton",
+                method: "prop",
+                args: ["disabled", "{change}.value"],
+                namespace: "uploadButtonDisabledChange"
+            }
```

Comment:
May be able to collapse this down into a single listener handler that listens to model changes on all of the specified paths and performs the necessary toggling based off the model change.

Reply:
As we discussed earlier today, this would require a model listener that listens to multiple paths (which is possible) but which also needs to know which DOM element it is going to be acting on.

Short of making the names of the model values consistently refer to the selector names and automatically extracting the latter from the former (which strikes me as a brittle and overcomplicated solution), I'm not sure there's a nice way to combine these into a single listener.

Reply:
Let's not worry about this for the current PR and we can revisit if needed later.